docker_daemon/check.py

```diff
@@ -644,6 +646,19 @@ def _report_net_metrics(self, container, tags):
             return
 
         proc_net_file = os.path.join(container['_proc_root'], 'net/dev')
+
+        try:
+            if container['Id'] in self.network_mappings:
+                networks = self.network_mappings[container['Id']]
+            else:
+                self.log.debug("Fetching network mapping for container %s" % container['Id'])
+                networks = self.docker_util.get_container_network_mapping(container)
+                self.network_mappings[container['Id']] = networks
+        except Exception as e:
+            # Revert to previous behaviour if the method is missing or failing
+            self.warning("Failed to build docker network mapping, using failsafe. Exception: {0}".format(e))
+            networks = {'eth0': 'bridge'}
```

Comment:
we can cache that to avoid future failures. If the network config changes and won't fail anymore, the cache will be invalidated anyway